nix/default.nix

```
@@ -302,14 +302,13 @@ in rec {
     exec ${pkgs.nodejs}/bin/npm start
   '';
 
-  services-req = [
-    elasticsearchStart
-    promStart
-    grafanaStart
-    monocleReplStart
-    monocleWebStart
-    monocleGhcid
-  ];
+  services-req =
+    [ elasticsearchStart monocleReplStart monocleWebStart monocleGhcid ];
+
+  monitoring-shell = hsPkgs.shellFor {
+    packages = p: [ p.monocle ];
+    buildInputs = [ promStart grafanaStart ];
+  };
```

This does not need hsPkgs, it would be better to use `mkShell` directly:

```suggestion
  monitoring-shell = pkgs.mkShell {
    buildInputs = [ promStart grafanaStart ];
  };
```

But then, now that there is a flake, it would be even better to use an app entry like: https://github.com/change-metrics/monocle/blob/7a6fb4c3c3e30d85d375b1f7f10a57e5b2d78a71/flake.nix#L46-L49
 by setting the `program` attribute to the `promStart` and `grafanaStart` value. That way we could simply do `nix run .#prometheus` and `nix run .#grafana`.

---

Good idea! see the followup patch.